bindings/java/src/main/native/context.cc

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+#include "context.h"
+#include "request_util.h"
+
+ucs_status_t context::ref_context() {
+    ucs_status_t status = UCS_OK;
+    if (ucp_context == nullptr) {
```

Comment:
do we need some thread safety here?

Reply:
I think you're right. There is a possible race condition. Should I wrap the if() statement with a c++ lock?

Reply:
yes, and the refcount should be inside too (so it does not have to be atomic any more)
same thing in deref